### PR TITLE
feat(userconfig): required fields

### PR DESCRIPTION
**Pull request by Serpentiel**
## About this change—what it does

adds support for required fields to userconfig generator

## Why this way

to have required fields in place

### CODE REVIEW

File: internal/schemautil/userconfig/handle.go [hunk 1]
```diff
@@ -161,13 +170,19 @@ func handleArrayProperty(n string, p map[string]interface{}, t string) (map[stri
...
 			}
 
-			e, err = handleArrayOfAggregateTypeProperty(ipa)
+			req := map[string]struct{}{}
+
+			if sreq, ok := ia["required"].([]interface{}); ok {
+				req = SliceToKeyedMap(sreq)
+			}
+
+			e, err = handleArrayOfAggregateTypeProperty(ipa, req)
```
Comment (byashimov) [inline internal/schemautil/userconfig/handle.go:179]: Please translate: what are iof, ipa, ia?

File: internal/schemautil/userconfig/handle.go [hunk 2]
```diff
@@ -219,9 +234,15 @@ func handleAggregateTypeProperty(
...
+	req := map[string]struct{}{}
+
+	if sreq, ok := p["required"].([]interface{}); ok {
+		req = SliceToKeyedMap(sreq)
+	}
+
 	switch at {
 	case "object":
-		v, err := handleObjectProperty(n, p, t)
+		v, err := handleObjectProperty(n, p, t, req)
```
Comment (byashimov) [inline internal/schemautil/userconfig/handle.go:245]: If `handleObjectProperty` handles just one property, then it is possible to this:
```go
requiredList, ok := p["required"].([]interface{})

// This is mandatory
if !ok {
    return fmt.Errorf("...")
}

handleObjectProperty(a, b, c, slices.Contains(requiredList, fieldName)
```

File: internal/schemautil/userconfig/handle.go (hunk 1 above)
Comment (Serpentiel) [reply internal/schemautil/userconfig/handle.go:179]: this is planned to be changed in future, after we release v4, but here are the abbreviations used:

`iof` is item one of, `ipa` is item properties asserted, `ia` is item asserted

File: internal/schemautil/userconfig/handle.go (hunk 2 above)
Comment (Serpentiel) [reply internal/schemautil/userconfig/handle.go:245]: it handles multiple properties in downstream call of `convertPropertiesToSchemaMap`

**Comment by byashimov** [comment]
lgtm

**Pull request merged**